File: SWEA/D3/1206. ［S／W 문제해결 기본］ 1일차 － View/［S／W 문제해결 기본］ 1일차 － View.py
for test_case in range(1, 11): 
    N = int(input())
    building_list = list(map(int, input().split()))
    answer = 0
    
    for i in range(2, N - 2):
        lhighest = max(building_list[i - 1], building_list[i - 2])
        rhighest = max(building_list[i + 1], building_list[i + 2])
        best_highest = max(lhighest, rhighest) 
        if best_highest < building_list[i]:
            answer += building_list[i] - best_highest
                
    print(f"#{test_case} {answer}")

File: SWEA/D3/1206. ［S／W 문제해결 기본］ 1일차 － View/［S／W 문제해결 기본］ 1일차 － View.cpp
#include <iostream>
#include <algorithm>
#include <vector>

using namespace std;

int main()
{
	for (int test_case = 1; test_case < 11; test_case++)
	{
		int N;
		cin >> N;

		vector<int> v(N);
		for (int i = 0; i < N; i++)
		{
			cin >> v[i];
		}

		int answer = 0;

		for (int i = 2; i < N-2; i++)
		{
			int L_best = max(v[i - 1], v[i - 2]);
			int R_best = max(v[i + 1], v[i + 2]);
			int T_best = max(L_best, R_best);

			if (v[i] > L_best && v[i] > R_best)
				answer += v[i] - T_best;
		}
		cout << "#" << test_case << " " << answer << "\n";
	}
}